src/index.js

```diff
@@ -82,7 +83,16 @@ function configureInterceptors(instance, url, { username, password } = {}) {
 
   instance.interceptors.response.use(res => {
     return res.data;
+  }, err => {
+    if (!(err instanceof Error) && typeof err === "object") {
+      return Promise.reject(errorify(err));
+    }
+    return Promise.reject(err);
```

Comment:
I was proposing that every error we throw should be an http error, i.e. if something throws a standard error we should still convert it anyway. But not mandatory by any means, i just thought it might be a good idea? I'll leave it to you to decide.